src/main/scala/com/gradle/internal/BuildScanTransformer.scala

```diff
@@ -0,0 +1,377 @@
+        sbt.url(s"$goServerUrl/tab/build/detail/$pipelineName/$pipelineNumber/$stageName/$stageNumber/$jobName")
+      }
+      val buildLink = goCdBuild.map(("GoCD build", _)).orElse(goServerUrl.map(("GoCD", _)))
+
+      val ops = Seq(
+        (bs: BuildScan) => bs.value("CI provider", "GoCD"),
+        ifDefined(buildLink) { case (bs, (label, url)) => bs.link(label, url) },
+        ifDefined(pipelineName)(withCustomValueAndSearchLink(_, "CI pipeline", _)),
+        ifDefined(jobName)(withCustomValueAndSearchLink(_, "CI job", _)),
+        ifDefined(stageName)(withCustomValueAndSearchLink(_, "CI stage", _))
+      )
+      Function.chain(ops)
+    }
+  }
+
+  private val captureAzurePipelines: BuildScan => BuildScan = {
+    if (!isAzurePipelines) identity
+    else {
+      val azureServerUrl = env.env[URL]("SYSTEM_TEAMFOUNDATIONCOLLECTIONURI")
+      val buildId = env.env[String]("BUILD_BUILDID")
+      val buildUrl = for {
+        azureServerUrl <- azureServerUrl
+        azureProject <- env.env[String]("SYSTEM_TEAMPROJECT")
+        buildId <- buildId
+      } yield sbt.url(s"$azureServerUrl$azureProject/_build/results?buildId=$buildId")
+      val buildLink = buildUrl.map(("Azure Pipelines build", _)).orElse(azureServerUrl.map(("Azure Pipelines", _)))
+
+      val ops = Seq(
+        (bs: BuildScan) => bs.value("CI provider", "Azure Pipelines"),
+        ifDefined(buildLink) { case (bs, (label, url)) => bs.link(label, url) },
+        ifDefined(buildId)(_.value("CI build number", _))
+      )
+      Function.chain(ops)
+    }
+  }
+
+  private val captureBuildkite: BuildScan => BuildScan = {
+    if (!isBuildkite) identity
+    else {
+      val prSource = for {
+        repository <- env.env[String]("BUILDKITE_PULL_REQUEST_REPO")
+        webRepoUri <- toWebRepoUri(repository)
+        prNumber <- env.env[String]("BUILDKITE_PULL_REQUEST")
+      } yield sbt.url(s"$webRepoUri/pull/$prNumber")
+
+      val ops = Seq(
+        (bs: BuildScan) => bs.value("CI provider", "Buildkite"),
+        ifDefined(env.env[URL]("BUILDKITE_BUILD_URL"))(_.link("Buildkite build", _)),
+        ifDefined(env.env[String]("BUILDKITE_COMMAND"))(withCustomValueAndSearchLink(_, "CI command", _)),
+        ifDefined(env.env[String]("BUILDKITE_BUILD_ID"))(_.value("CI build number", _)),
+        ifDefined(prSource)(_.link("PR source", _))
+      )
+      Function.chain(ops)
+    }
+  }
+
+  private val captureCiMetadata: BuildScan => BuildScan = {
+    val ops = Seq(
+      captureJenkinsOrHudson,
+      captureTeamCity,
+      captureCircleCi,
+      captureBamboo,
+      captureGitHubActions,
+      captureGitLab,
+      captureTravis,
+      captureBitrise,
+      captureGoCd,
+      captureAzurePipelines,
+      captureBuildkite
+    )
+    Function.chain(ops)
+  }
+
+  private val captureGitMetadata: BuildScan => BuildScan = {
+    if (!isGitInstalled) identity
+    else {
+      val gitRepo = execAndGetStdOut("git", "config", "--get", "remote.origin.url")
```

Comment:
for my understanding, is executing these git commands blocking the build?

Reply:
Yes, the sbt develocity plugin doesn't have a background handle like Gradle, so the processing is done on main thread.